pkg/utils/net-attach-def.go

```diff
@@ -152,14 +152,41 @@ func CreateNetworkStatuses(r cnitypes.Result, networkName string, defaultNetwork
 	// Same for DNS
 	v1dns := convertDNS(result.DNS)
 
+	// Check for a gateway-associated interface, we'll use this later if we did to mark as the default.
+	gatewayInterfaceIndex := -1
+	if defaultNetwork {
+		for _, ipConfig := range result.IPs {
+			if ipConfig.Gateway != nil && ipConfig.Interface != nil {
+				// Keep the index of the first interface that has a gateway
+				gatewayInterfaceIndex = *ipConfig.Interface
+				break
+			}
+		}
+	}
+
 	// Initialize NetworkStatus for each container interface (e.g. with sandbox present)
 	indexOfFoundPodInterface := 0
 	foundFirstSandboxIface := false
+	didSetDefault := false
 	for i, iface := range result.Interfaces {
 		if iface.Sandbox != "" {
+			isDefault := false // default to false by default
```

Comment:
nit: I feel the comment is a bit redundant

Reply:
+1